library/choose_disk.py

```diff
@@ -0,0 +1,436 @@
+    If no count is provided, a default value of 1 is taken.
+    '''
+    final_list = {}
+    for disk in lookup_disks:
+        infinite = False
+        count = 0
+        if 'count' not in lookup_disks[disk]:
+            fatal("disk '%s' should have a 'count' value defined" % disk)
+        if 'count' in lookup_disks[disk]:
+            count = lookup_disks[disk]['count']
+            del lookup_disks[disk]['count']
+
+        if '*' in str(count).strip():
+            infinite = True
+            count = 1
+
+        for n in range(0, int(count), 1):
+            final_list["%s_%03d" % (disk, n)] = lookup_disks[disk]
+            if infinite is True:
+                final_list["%s_%03d" % (disk, n)]["infinite"] = "1"
+
+    return final_list
+
+
+def is_ceph_disk(partition):
+    '''
+    Check if a parition is used by ceph
+    '''
+    try:
+        stdout = subprocess.check_output(["lsblk", "-no", "PARTLABEL", "%s" % partition])
+        if "ceph data" in stdout:
+            return True
+    except:
+        pass
+
+    return False
+
+
+def select_only_free_devices(physical_disks):
+    ''' Don't keep that have partitions '''
+    selected_devices = {}
+    info('Detecting free devices')
+    for physical_disk in sorted(physical_disks):
+        ceph_disk = False
+        current_physical_disk = physical_disks[physical_disk]
+
+        # Don't consider devices that doesn't have partitions
+        if 'partitions' not in current_physical_disk:
+            info(' Ignoring %10s : Device doesnt support partitioning' % physical_disk)
+            continue
+        # Don't consider the device if partition list is not empty,
+        if len(current_physical_disk['partitions']) > 0:
+            for partition in current_physical_disk['partitions']:
+                if is_ceph_disk("/dev/" + partition):
+                    ceph_disk = True
+
+            if ceph_disk is False:
+                info(' Ignoring %10s : Device have exisiting partitions' % physical_disk)
+                continue
+
+        selected_devices[physical_disk] = physical_disks[physical_disk]
+        selected_devices[physical_disk]['bdev'] = '/dev/' + physical_disk
+
+        if ceph_disk is True:
+            selected_devices[physical_disk]['ceph'] = 1
+            info(' Adding   %10s : Ceph disk detected' % physical_disk)
+        else:
+            info(' Adding   %10s : %s' % (physical_disk, selected_devices[physical_disk]['bdev']))
+
+    return selected_devices
+
+
+def get_block_devices_persistent_name(physical_disks):
+    ''' Replace the short name (sda) by the persistent naming 'by-id' '''
+    directory = "/dev/disk/by-id/"
+
+    info('Finding persistent disks name')
+    # If the directory doesn't exist, reports the list as-is
+    if not os.path.isdir(directory):
+        info(' Cannot open %s' % directory)
+        return physical_disks
+
+    final_list = {}
+    matching_list = {}
```

Comment:
That was more about the semantic than the data structure type. I renamed them properly to avoid any confusion.